service/src/main/java/org/cbioportal/genome_nexus/service/cached/BaseCachedVariantAnnotationFetcher.java

```diff
@@ -0,0 +1,102 @@
+    protected ResourceTransformer<VariantAnnotation> transformer;
+    protected Integer maxPageSize;
+
+    public BaseCachedVariantAnnotationFetcher(String collection,
+                                             VariantAnnotationRepository repository,
+                                             Class<VariantAnnotation> type,
+                                             ExternalResourceFetcher<VariantAnnotation> fetcher,
+                                             ResourceTransformer<VariantAnnotation> transformer)
+    {
+        super(collection, repository, type, fetcher, transformer, Integer.MAX_VALUE);
+    }
+
+    public BaseCachedVariantAnnotationFetcher(String collection,
+                                             VariantAnnotationRepository repository,
+                                             Class<VariantAnnotation> type,
+                                             ExternalResourceFetcher<VariantAnnotation> fetcher,
+                                             ResourceTransformer<VariantAnnotation> transformer,
+                                             Integer maxPageSize)
+    {
+        super(collection, repository, type, fetcher, transformer, maxPageSize);
+    }
+   
+    @Override
+    protected String extractId(VariantAnnotation instance)
+    {
+        return instance.getVariantId();
+    }
+
+    @Override
+    protected String extractId(DBObject dbObject)
+    {
+        return (String)dbObject.get("input");
+    }
+
+    @Override 
+    public VariantAnnotation fetchAndCache(String id) throws ResourceMappingException
+    {
+        VariantAnnotation variantAnnotation = null;
+        try {
+            variantAnnotation = super.fetchAndCache(id);
+            if (variantAnnotation == null) {
+                variantAnnotation = new VariantAnnotation(id);
+            } else {
+                variantAnnotation.setSuccessfullyAnnotated(true);
+            }
+        } catch (Exception e) {
+            return new VariantAnnotation(id);
+        }
+        return variantAnnotation;
+    }
+    
+    @Override
+    public List<VariantAnnotation> fetchAndCache(List<String> ids) throws ResourceMappingException
+    {
+        Map<String, VariantAnnotation> variantResponse = super.constructFetchedMap(ids); 
+        for (String variantId : variantResponse.keySet()) {
+            if (variantResponse.get(variantId) == null) {
+                VariantAnnotation variantAnnotation = new VariantAnnotation(variantId);
+                variantResponse.put(variantId, variantAnnotation);
+            } else {
+                variantResponse.get(variantId).setSuccessfullyAnnotated(true);
+            }
+        }
+        // some util function to return response (values) with duplicates
+        // need to pass in a list representing indexes of original request
+        List<VariantAnnotation> values = new ArrayList();
+        for (String id : ids) {
```

Comment:
use the order of the requests to fill the response correctly ... and creates duplicate entries when duplicated in input.
It seems like this is done - maybe the comment above does not need to refer to a util function.